src/modules/colorPicker/ColorPickerUI/Controls/ColorPickerControl.xaml.cs

```diff
@@ -288,22 +288,40 @@ private void ValueGradientSlider_ValueChanged(object sender, RoutedPropertyChang
 
         private void HexCode_TextChanged(object sender, TextChangedEventArgs e)
         {
+            bool insertHexTextHashtag = false;
             var newValue = (sender as TextBox).Text;
 
-            // support hex with 3 and 6 characters
-            var reg = new Regex("^#([0-9A-Fa-f]{3}){1,2}$");
+            // support hex with 3 or 6 characters and with or without #
```

Comment:
@stefansjfw , @niels9001 
Here we accept three values input but we can't use it correctly. We interpret three values input as the first three values of a six values input.

Should we remove the matching of three digits input or should we add an converter? 

I have opened an issue here:

Reply:
I personally suggest to block three digits input untils the converter is implemted and the issue is fixed.

Reply:
I'm not experienced with regex so @stefansfjw can better judge it.

FYI if Color Picker moves to WinUI3 we can remove all of this custom code and use the default WinUI component.

Reply:
I closed this pr. I will create a new one in the next few weeks for fixing all bugs together. Otherwise we have to change code twice which is imo not necessary